File: src/main/java/com/server/ordering/domain/dto/response/RestaurantPreviewWithDistanceDto.java
package com.server.ordering.domain.dto.response;

import lombok.*;

import static lombok.AccessLevel.PROTECTED;

@Getter @Setter
@NoArgsConstructor(access = PROTECTED)
public class RestaurantPreviewWithDistanceDto extends RestaurantPreviewDto {

    private int distanceMeter;

    public RestaurantPreviewWithDistanceDto(Long restaurantId, String restaurantName, String profileImageUrl, String backgroundImageUrl, int distanceMeter) {
        super(restaurantId, restaurantName, profileImageUrl, backgroundImageUrl);
        this.distanceMeter = distanceMeter;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/dto/response/RestaurantPreviewWithDistanceDto.kt
package com.byeoru.ordering_server.domain.dto.response

data class RestaurantPreviewWithDistanceDto(val distanceMeter: Int,
                                            val restaurantId: Long,
                                            val restaurantName: String,
                                            val profileImageUrl: String,
                                            val backgroundImageUrl: String,
                                            val representativeMenus: MutableList<String> = ArrayList()) {

    fun addRepresentativeFoodName(foodName: String) = representativeMenus.add(foodName)
}
